commands/analyze.go

```diff
@@ -39,109 +39,142 @@ func init() {
 
 func analyze(inDb string, configFile string) error {
 	res := resources.InitResources(configFile)
-	var toRunDirty []string
-	var toRun []string
+	var toRunDirty []database.RITADatabase
+	var toRun []database.RITADatabase
 
 	// Check to see if we want to run a full database or just one off the command line
 	if inDb == "" {
-		res.Log.Info("Running analysis against all databases")
-		toRunDirty = append(toRun, res.MetaDB.GetUnAnalyzedDatabases()...)
-	} else {
-		toRunDirty = append(toRun, inDb)
-	}
+		res.Log.Info("Running analysis against all unanalyzed databases")
+		unanalyzedDBs, err := res.DBIndex.GetUnanalyzedDatabases()
+		if err != nil {
+			errStr := fmt.Sprintf("Error: failed to list unanalyzed databases: %s", err.Error())
+			res.Log.Error(errStr)
+			return cli.NewExitError(errStr, 255)
+		}
 
-	// Check for problems
-	for _, possDB := range toRunDirty {
-		info, err := res.MetaDB.GetDBMetaInfo(possDB)
+		//ensure the databases have finished being imported
+		for i := range unanalyzedDBs {
+			if unanalyzedDBs[i].ImportFinished() {
+				toRunDirty = append(toRunDirty, unanalyzedDBs[i])
+			}
+		}
```

Comment:
This is the big change in this PR. We only analyze databases which have finished importing.